docs/library/common/ad_dds/index.rst

```diff
@@ -0,0 +1,331 @@
+The DDS_INIT value will be used by the phase accumulator as a starting point in
+other words as a phase offset.
+
+The offset can be determined from the phase accumulator capacity 0 to
+:math:`2^{phaseDW}` represents (0° - 360°).
+
+e.g., for 90°. :math:`init = (90 * 2^{phaseDW})/360`.
+
+FREQUENCY - DDS_INCR
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+
+The value can be calculated by:
+
+.. math::
+
+   INCR = \frac{(f_{out} * 2^{phaseDW}) * clkratio}{f_{if}}
+
+Where:
+
+#. f_out is the generated output frequency
+#. phaseDW (``DDS_PHASE_DW``) value can be found in ``CHAN_CNTRL_1`` in case
+   ``DDS_PHASE_DW`` is not eq. 16
+#. f_if is the frequency of the digital interface
+#. clock_ratio is the ratio between the sampling clock and the interface clock.
+
+
+STRUCTURE
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Below is the hierarchical structure of the modules.
+
+.. code::
+
+  * ad_dds ->
+    * ad_dds_2 ->
+      * ad_dds_1 ->
+        * ad_dds_sine_*
+
+ad_dds
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:git-hdl:`ad_dds <library/common/ad_dds.v>`
+is the main module, it contains the phase accumulators and the phase to
+amplitude converters.
+
+.. image:: dds_dual_tone.svg
+
+ad_dds_2
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:git-hdl:`ad_dds_2 <library/common/ad_dds_2.v>`
+contains two phase to amplitude converters. The resulting wave-forms will be
+summed. The resulting wave-form must have a maximum amplitude level of 0x8000
+-1. When only one tone is desired both tones must have the same frequency word
+and shift, there is no constraint for the amplitude, but if is equal for both
+
+ad_dds_1
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:git-hdl:`ad_dds_1 <library/common/ad_dds_1.v>`
+contains the phase to amplitude converter and an "amplifier". The phase to
+amplitude converter is always generating a full scale sine-wave. Because the
+data format is two's complement, for a 16 bit data angle, min value will be
+-(2^16)/2 and max (2^16)/2-1.
+
+.. image:: ad_dds_1.svg
+
+ad_dds_sine_*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:git-hdl:`ad_dds_sine <library/common/ad_dds_sine.v>`
+and
+:git-hdl:`ad_dds_sine_cordic <library/common/ad_dds_sine_cordic.v>`
+are the available phase to amplitude converters.
+
+CLOCK RATIO
+-------------------------------------------------------------------------------
+
+The clock ratio (number data paths processed in parallel) instantiates more DDS
+logic, but it it is controlled by the same register map for all
+CLOCK_RATIO/DATA_PATH parameters.
+
+Where is the CLOCK_RATIO > 1? This scenario can be found in every high speed DAC
+design. It is required because the FPGA fabric can't work at the same speed than
+that of a high speed converter(typically > 250M).
```

Comment:
```suggestion
design. It is required (CLOCK_RATIO > 1) because the FPGA fabric can't work at the same speed as
that of the high-speed converter (typically > 250M).
```